### PR TITLE
Return os.ErrNotExists from map deletes.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Avoids log spam in this expected case.

Calling code was already expecting the new error.

## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
In BPF mode, fix spurious "Failed to run bpftool" logs.
```

### CODE REVIEW

File: bpf/maps.go [hunk 1]
```diff
@@ -222,6 +222,15 @@ func (b *PinnedMap) Delete(k []byte) error {
 	cmd := exec.Command("bpftool", args...)
 	out, err := cmd.CombinedOutput()
 	if err != nil {
+		if strings.Contains(string(out), "No such file or directory") {
+			// Check that the error wasn't about the map as a whole being missing.
+			if _, statErr := os.Stat(b.versionedFilename()); statErr == nil {
```
Comment (tomastigera) [inline bpf/maps.go:227]: The error already tells you what sort of ENOENT it is, doesnt it?

```
[root@suez /]# bpftool -j -p map delete pinned /sys/fs/bpf/tc/globals/cali_v4_nat_aff key 01 02 03 04 05 06 07 01 02 03 04 05 06 07 01 01
{
    "error": "delete failed: No such file or directory"
}
[root@suez /]# bpftool -j -p map delete pinned /sys/fs/bpf/tc/globals/cali_v4_nat_aff123 key 01 02 03 04 05 06 07 01 02 03 04 05 06 07 01 01
{
    "error": "bpf obj get (/sys/fs/bpf/tc/globals/cali_v4_nat_aff123): No such file or directory"
}
```

File: bpf/maps.go [hunk 2]
```diff
@@ -222,6 +222,15 @@ func (b *PinnedMap) Delete(k []byte) error {
...
+				// We expect failures due to the entry being missing, return a dedicated error
+				// and avoid logging a scary warning.
+				logrus.WithField("k", k).Debug("Item didn't exist.")
+				return os.ErrNotExist
+			}
+		}
 		logrus.WithField("out", string(out)).Error("Failed to run bpftool")
 	}
 	return err
```
Comment (tomastigera) [inline bpf/maps.go:236]: What is the error if the whole map does not exist? Not the os.Err ?

File: bpf/maps.go (hunk 1 above)
Comment (fasaxc) [reply bpf/maps.go:227]: Didn't spot that, will respin

File: bpf/maps.go (hunk 2 above)
Comment (fasaxc) [reply bpf/maps.go:231]: It'll be an exec error

**Pull request merged**